icu4c/source/common/uloc_tag.cpp

```diff
@@ -1110,6 +1110,19 @@ _appendRegionToLanguageTag(const char* localeID, icu::ByteSink& sink, UBool stri
     }
 }
 
+static void _sortVariants(VariantListEntry* first) {
+    for (VariantListEntry* var1 = first; var1 != NULL; var1 = var1->next) {
+        for (VariantListEntry* var2 = var1->next; var2 != NULL; var2 = var2->next) {
+            // Swap var1->variant and var2->variant.
+            if (strcmp(var1->variant, var2->variant) > 0) {
```

Comment:
In the ICU4C library code, we want to use uprv_str...() functions not raw str...().

In this case, we would also get inconsistent sort orders for EBCDIC vs. ASCII, which would make your test cases fail on an EBCDIC machine, so actually the right one to use is uprv_compareInvCharsAsAscii(s1, s2) from internal uinvchar.h. (EBCDIC sorts digits after letters.)

(This is not a concern in Java which uses Unicode strings.)

Reply:
PTAL adddressed!